src/App.tsx

```diff
@@ -1,223 +1,85 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorMessage } from './components/ErrorMessage';
+import { TodoList } from './components/TodoList';
+import { FilterTypes } from './types/FilterTypes';
+import { Footer } from './components/Footer';
+import { TodoField } from './components/TodoField';
+
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState(false);
+  const [loadError, setLoadError] = useState(false);
+  const [filterType, setFilterType] = useState<FilterTypes>(FilterTypes.all);
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const filteredTodos = todos.filter(todo => {
+    switch (filterType) {
+      case FilterTypes.active:
+        return !todo.completed;
+
+      case FilterTypes.completed:
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  });
```

Comment:
It's just a recommendation, not a change request. I advice you to read about useMemo.
https://beta.reactjs.org/apis/react/useMemo

In a nutshell, when either props or state of React component changes it makes the component rerender. And new value for filteredTodos is calculated on every render as for now. It may be bad for slow devices as UX may be laggy. This hook helps to optimize such things.